h/notification/reply_template.py

```diff
@@ -26,55 +26,47 @@
 
 def create_template_map(request, reply, parent):
     document_title = ''
-    if 'document' in reply:
-        document_title = reply['document'].get('title', '')
+    if reply.document:
+        document_title = reply.document.title
 
     if document_title is '':
-        document_title = parent['uri']
+        document_title = parent.target_uri
 
-    parent_user = user_name(parent['user'])
-    reply_user = user_name(reply['user'])
+    parent_user = user_name(parent.userid)
+    reply_user = user_name(reply.userid)
 
     token = request.registry.notification_serializer.dumps({
         'type': REPLY_TYPE,
-        'uri': parent['user'],
+        'uri': parent.userid,
     })
     unsubscribe = request.route_url('unsubscribe', token=token)
 
     return {
         'document_title': document_title,
-        'document_path': parent['uri'],
-        'parent_text': parent.get('text', ''),
+        'document_path': parent.target_uri,
+        'parent_text': parent.text,
         'parent_user': parent_user,
-        'parent_timestamp': format_timestamp(parent['created']),
-        'parent_user_profile': user_profile_url(request, parent['user']),
-        'parent_path': standalone_url(request, parent['id']),
-        'reply_text': reply['text'],
+        'parent_timestamp': format_timestamp(parent.created),
+        'parent_user_profile': user_profile_url(request, parent.userid),
+        'parent_path': standalone_url(request, parent.id),
+        'reply_text': reply.text,
         'reply_user': reply_user,
-        'reply_timestamp': format_timestamp(reply['created']),
-        'reply_user_profile': user_profile_url(request, reply['user']),
-        'reply_path': standalone_url(request, reply['id']),
+        'reply_timestamp': format_timestamp(reply.created),
+        'reply_user_profile': user_profile_url(request, reply.userid),
+        'reply_path': standalone_url(request, reply.id),
         'unsubscribe': unsubscribe
     }
 
 
-def format_timestamp(timestamp):
-    # Currently we cut the UTC format because time.strptime has problems
-    # parsing it, and of course it'd only correct the backend's timezone
-    # which is not meaningful for international users. This trims the
-    # timezone in the format +00:00.
-    timestamp = re.sub(r'\+\d\d:\d\d$', '', timestamp)
-    timestamp_format = '%Y-%m-%dT%H:%M:%S.%f'
-    parsed = datetime.strptime(timestamp, timestamp_format)
-
+def format_timestamp(timestamp, now=datetime.now):
```

Comment:
I know you're just preserving behaviour, but this should definitely be `utcnow`.